tests/unit/RequirementsTest.php

```diff
@@ -56,6 +56,13 @@ public function setUp() {
 		$this->context->logger = new FaceRecognitionLogger($logger);
 	}
 
+	public function testModelInstalled() {
+		$config = $this->createMock(IConfig::class);
+		$defaultModel = AddDefaultFaceModel::DEFAULT_FACE_MODEL_ID;
+		$modelInstalled = intval($config->getAppValue('facerecognition', 'model', AddDefaultFaceModel::DEFAULT_FACE_MODEL_ID));
```

Comment:
@stalker314314 
Last travis says:
 * `Failed asserting that 0 matches expected 1.`

So, https://github.com/matiasdelellis/facerecognition/blob/master/lib/Migration/AddDefaultFaceModel.php is not executed??

Reply:
if it was not installed, all tests would fail? And `getAppValue` would return provided default value, which is `AddDefaultFaceModel::DEFAULT_FACE_MODEL_ID`, so it should match even if repair step is not run:)

There is this line in Travis:
```
Build done. You can enable the application in Nextcloud.
./vendor/bin/phpunit --coverage-clover clover.xml -c phpunit.xml
```

Do Travis need to enable app? Only after app is enabled, upgrade step should be run?

Weird...

Reply:
I never used `$config` as mock in unit tests, only in integration tests. Are you sure mock object is not just returning default value for int type, which is 0?